cmd/service-controller/service_sync.go

```diff
@@ -128,7 +128,14 @@ func (c *Controller) ensureServiceInterfaceDefinitions(origin string, serviceInt
 
 	c.heardFrom[origin] = time.Now()
 
+	policy := client.NewClusterPolicyValidator(c.vanClient)
+
 	for _, def := range serviceInterfaceDefs {
+		res := policy.ValidateImportService(def.Address)
```

Comment:
If we move the logic to the service controller itself, lets say under `updateServiceBindings`, then there is
no need to handle it here, as part of the service sync.

The reason why I added it here, is to ignore a not authorized service from existing in a remote site at all.
I know that this will generate lots of events every time services are synced and ignored, but the intention is
actually to see that the service creation was rejected.

What do you see as a benefit of moving this somewhere else, like the controller itself?

Reply:
An aim of the policy is to give the cluster owner assurance about what will and won't be exposed to or by skupper. I think we need to enforce the services that are allowed in the service controller to ensure that someone cannot workaround the restrictions by directly editing skupper-services.

If we do that, then there is actually no way that a service advertised by some other site through service-sync is exposed to workloads in the cluster except through allowed service names. That being the case, I wouldn't see the need to prevent the advertised services being locally recorded.

Reply:
It works. But after removing this, `skupper status`  and `skupper service status` will list
the unauthorized service as well.

If we also keep this one, then the unauthorized service won't be listed, otherwise, I can
add an extra validation to: `skupper service status` and `skupper network status`
so that unauthorized services are listed as "not authorized". In this case then the console/services endpoint
might need some work either.

Reply:
This is perhaps something to discuss a little more. Personally, I like the approach of updating the skupper-services definition, but flagging that the service is not creating due to policy. However others may prefer the simplicity of not allowing them in in the first place.

Reply:
@grs Removed this validation from the service sync. Services will show up now and, if not authorized,
the respective kubernetes service won't be created and `skupper service status` as well as 
`skupper network status` will display a `not authorized` message.
@nluaces pls review again.

Reply:
> This is perhaps something to discuss a little more. Personally, I like the approach of updating the skupper-services definition, but flagging that the service is not creating due to policy. However others may prefer the simplicity of not allowing them in in the first place.

Initially my preference was no to let them in at all, but after your brought that point of manually editing the skupper-services, I had a second thought and actually I believe having the services showing up as not-authorized makes more sense, as it will be clear to users that skupper is being notified about those services and they will know the reason why they do not work.

If those services are not listed at all, then users won't have any clue why, unless they query for internal skupper events.